cupy_setup_build.py

```diff
@@ -356,6 +356,13 @@ def preconfigure_modules(compiler, settings):
         status = 'No'
         errmsg = []
 
+        if module['name'] == 'cub':
+            # for <cupy/complex.cuh>
+            cupy_header = os.path.join(
+                os.path.dirname(os.path.realpath(__file__)),
+                'cupy/core/include')
+            settings['include_dirs'].append(cupy_header)
```

Comment:
Hi @anaruse, is this needed? I thought the CuPy header path is already added in `get_compiler_setting()` (a few lines above).

By the way, I intend to support CUDA 11 CUB in #2584. Could you comment my approach there? Thanks!

Reply:
I'm sorry I couldn't mention this until CUDA11 was officially released, but starting with CUDA11, CUB is now included in CUDA toolkit.

Reply:
@anaruse I think it's semi-public info now...I already knew this from the release announcements and PRs in [thrust/cub](https://github.com/thrust/cub/) 😇 So, from the implementation in [thrust/thrust](https://github.com/thrust/thrust/), I inferred that the CUB headers will be available in `/usr/local/cuda/include/cub/`, and made the changes accordingly.

Reply:
With CUDA 11.0, you don't need to set the CUB_PATH, so I modified `preconfigure_modules()` to also add the CuPy header path here regardless of the CUB_PATH setting. But I agree that adding CuPy header path here is not appropriate, so I'll make sure the include path is set correctly in `get_compiler_setting()` for CUDA 11.0 as well.

Reply:
BTW, I did not know that you knew this already :sweat:

Reply:
It's open now: 
> cub (High performance primitives for CUDA)

https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html
It's just very odd that this is the only mention of CUB throughout the whole documentation; no other references or standalone sections (ex: Thrust has its own section). I wouldn't know what it is if I had not knew it already...